javascript/ql/src/Security/CWE-116/IncompleteMultiCharacterSanitization.ql

```diff
@@ -56,31 +56,66 @@ DangerousPrefix getADangerousMatchedPrefix(EmptyReplaceRegExpTerm t) {
   not exists(EmptyReplaceRegExpTerm pred | pred = t.getPredecessor+() and not pred.isNullable())
 }
 
+private import semmle.javascript.security.performance.ReDoSUtil as ReDoSUtil
+
+/**
+ * Gets a char from a dangerous prefix that is matched by `t`.
+ */
+pragma[noinline]
+DangerousPrefixSubstring getADangerousMatchedChar(EmptyReplaceRegExpTerm t) {
+  t.isNullable() and result = ""
+  or
+  t.getAMatchedString() = result
+  or
+  exists(ReDoSUtil::CharacterClass cc |
+    cc = ReDoSUtil::getCanonicalCharClass(t) and
+    cc.matches(result) and
+    not cc.matches(">")
```

Comment:
I have tried to add a suggestion here with a comment that explains the reason for excluding `>` explicitly, but I can't do much better than `// needed to exclude /<[^<]*>/`, which seems a bit too ad hoc. Could we move the blacklist elsewhere, or formulate the general principle it implements.

Reply:
I think it's OK to just add a comment explaining why it's there. 
That comment would also explain that the `CharacterClass` matching is only relevant for the HTML-tags regexps. 
I'll try to get something in there. 

I don't think we can make the code much prettier without basically duplicating a bunch of the implementation.

Reply:
Ok. Let's keep it simple then. 👍🏻